src/js/components/Popup.js

```diff
@@ -12,32 +13,58 @@ const Popup = forwardRef((props, ref) => {
     }
   }
 
-  const serializedProps = serializeProps([
-    "loading",
-    "service",
-    "subdomain",
-    "projects",
-    "activities",
-    "schedules",
-    "timedActivity",
-    "serviceLastProjectId",
-    "userLastProjectId",
-    "serviceLastTaskId",
-    "userLastTaskId",
-    "fromDate",
-    "toDate",
-    "errorType",
-    "errorMessage",
-  ])(props)
+  const sendSerializedProps = useCallback(() => {
+    if (iFrameRef.current) {
+      const serializedProps = serializeProps([
+        "loading",
+        "service",
+        "subdomain",
+        "projects",
+        "activities",
+        "schedules",
+        "timedActivity",
+        "serviceLastProjectId",
+        "userLastProjectId",
+        "serviceLastTaskId",
+        "userLastTaskId",
+        "fromDate",
+        "toDate",
+        "errorType",
+        "errorMessage",
+      ])(props)
+
+      iFrameRef.current.contentWindow.postMessage(
+        {
+          type: "popup-serialized-props",
+          serializedProps,
+        },
+        "*",
+      )
+    }
+  }, [props])
+
+  const handleRequestSerializedProps = useCallback(
+    (event) => {
+      if (event.data.type === "popup-request-serialized-props") {
+        sendSerializedProps()
+      }
+    },
+    [sendSerializedProps],
+  )
+
+  useEffect(() => {
+    sendSerializedProps()
+    window.addEventListener("message", handleRequestSerializedProps)
+
+    return () => window.removeEventListener("message", handleRequestSerializedProps)
+  }, [props, iFrameRef.current])
```

Comment:
Daten werden nun über postMessage (sendSerializedProps) and das iFrame geschickt.